StripePayments/StripePayments/Source/PaymentHandler/STPPaymentHandler.swift

```diff
@@ -1525,12 +1525,11 @@ public class STPPaymentHandler: NSObject {
                         timeout: pollingBudget?.networkTimeout
                     ) { [self] paymentIntent, error in
                         guard let paymentIntent, error == nil else {
-                            // If we got an error retrieving the intent, retry if budget allows.
-                            // Note: This will only retry if we have a pollingBudget, which means it's a polling call.
-                            // We won't retry if it's the first call (no pollingBudget). Ideally we should retry
-                            // on the first call too, but this is a limitation to address in a future rewrite.
-                            if let pollingBudget, pollingBudget.canPoll {
-                                pollingBudget.pollAfter {
+                            // Retry if polling budget allows. For the first call (no polling budget), create a minimal
+                            // budget to allow one retry. This handles transient network errors.
+                            let effectivePollingBudget = pollingBudget ?? PollingBudget(startDate: Date(), duration: 1)
+                            if effectivePollingBudget.canPoll {
+                                effectivePollingBudget.pollAfter {
```

Comment:
Code looks fine, but is this what we want to do? Are we seeing a pattern of failures in this case?

Reply:
Also got a report here: https://stripe.slack.com/archives/C02TNKMM1HT/p1759507487356389?thread_ts=1758918760.012799&cid=C02TNKMM1HT

We rolled the flag back so no more impact but this resolves the issue.